src/write_restart.cpp

```diff
@@ -661,7 +661,7 @@ void WriteRestart::write_double(int flag, double value)
 }
 
 /* ----------------------------------------------------------------------
-   write a flag and a char string (including NULL) into restart file
```

Comment:
This is incorrect. It writes the 0 byte at the end of the string. This has nothing to do with nullptr

Reply:
how about this?

Reply:
@rbberger This comment is an example of several I saw where the original comment used NULL rather loosely. I believe 
`write a flag and a char string (including NULL) into restart file`
should have said something more like :
`write a flag and a char string (including the terminating null char) into restart file`

My bulk change converted the NULL to nullptr in comments as well as code. Which in this case was certainly wrong-er than the original. I believe that, in the large majority of comments, the conversion to nullptr was the right choice.

When I hand-applied my cleanup to changed comments, I tried to notice the places where the original comment had used NULL in this way and replace with a comment more like the second above. I assume @akohlmey may have done the same. But, as this comment shows, our eyeballing and hand-fixing did not necessarily catch and improve every comment that used NULL in this way.

I'm happy to make another pass over comments looking for this specific problem.

Reply:
Axel already fixed it